honeybee_radiance/properties/model.py

```diff
@@ -304,11 +431,62 @@ def load_properties_from_dict(data):
         if 'modifier_sets' in data['properties']['radiance'] and \
                 data['properties']['radiance']['modifier_sets'] is not None:
             for m_set in data['properties']['radiance']['modifier_sets']:
-                modifier_sets[m_set['identifier']] = \
-                    ModifierSet.from_dict_abridged(m_set, modifiers)
+                if m_set['type'] == 'ModifierSet':
+                    modifier_sets[m_set['identifier']] = ModifierSet.from_dict(m_set)
+                else:
+                    modifier_sets[m_set['identifier']] = \
+                        ModifierSet.from_dict_abridged(m_set, modifiers)
 
         return modifiers, modifier_sets
 
+    def _indoor_shades(self):
+        """Get a list of all indoor Shade objects in the model."""
+        host = self.host
+        child_shades = []
+        for room in host._rooms:
+            child_shades.extend(room._indoor_shades)
+            for face in room.faces:
+                child_shades.extend(face._indoor_shades)
+                for ap in face._apertures:
+                    child_shades.extend(ap._indoor_shades)
+                for dr in face._doors:
+                    child_shades.extend(dr._indoor_shades)
+        for face in host._orphaned_faces:
+            child_shades.extend(face._indoor_shades)
+            for ap in face._apertures:
+                child_shades.extend(ap._indoor_shades)
+            for dr in face._doors:
+                child_shades.extend(dr._indoor_shades)
+        for ap in host._orphaned_apertures:
+            child_shades.extend(ap._indoor_shades)
+        for dr in host._orphaned_doors:
+            child_shades.extend(dr._indoor_shades)
+        return child_shades
+    
+    def _outoor_shades(self):
+        """Get a list of all outdoor Shade objects in the model."""
+        host = self.host
+        child_shades = []
+        for room in host._rooms:
+            child_shades.extend(room._outdoor_shades)
+            for face in room.faces:
+                child_shades.extend(face._outdoor_shades)
+                for ap in face._apertures:
+                    child_shades.extend(ap._outdoor_shades)
+                for dr in face._doors:
+                    child_shades.extend(dr._outdoor_shades)
+        for face in host._orphaned_faces:
```

Comment:
Any specific reason that you use the private property when you get them from host? Shouldn't we just use the public method?

Reply:
Sure. I did it to avoid the casting from the internal value (a list) to the public one (a tuple). But the extra time/memory it takes to do this casting once shouldn't make a difference. I'll change it back.

Reply:
Actually, I see that these hidden methods get called a lot and all of the casting to tuples is going to eat up time and memory. I will move them over into honeybee-core if these methods remain unchanged after we implement dynamic geometry. I was thinking that these methods might be temporary while I still don't understand if shades assigned to an aperture should be a part of the aperture or indoor/outdoor geometry.